Core/include/Acts/Seeding/SeedFinderOrthogonal.ipp

```diff
@@ -0,0 +1,489 @@
+      if (deltaCotTheta2 - error2 > 0) {
+        deltaCotTheta = std::abs(deltaCotTheta);
+        // if deltaTheta larger than the scattering for the lower pT cut, skip
+        error = std::sqrt(error2);
+        dCotThetaMinusError2 =
+            deltaCotTheta2 + error2 - 2 * deltaCotTheta * error;
+        // avoid taking root of scatteringInRegion
+        // if left side of ">" is positive, both sides of unequality can be
+        // squared
+        // (scattering is always positive)
+
+        if (dCotThetaMinusError2 > scatteringInRegion2) {
+          continue;
+        }
+      }
+
+      float dU = lt.U - Ub;
+
+      // A and B are evaluated as a function of the circumference parameters
+      // x_0 and y_0
+      float A = (lt.V - Vb) / dU;
+      float S2 = 1. + A * A;
+      float B = Vb - A * Ub;
+      float B2 = B * B;
+      // sqrt(S2)/B = 2 * helixradius
+      // calculated radius must not be smaller than minimum radius
+      if (S2 < B2 * m_config.minHelixDiameter2) {
+        continue;
+      }
+      // 1/helixradius: (B/sqrt(S2))*2 (we leave everything squared)
+      float iHelixDiameter2 = B2 / S2;
+      // calculate scattering for p(T) calculated from seed curvature
+      float pT2scatter = 4 * iHelixDiameter2 * m_config.pT2perRadius;
+      // if pT > maxPtScattering, calculate allowed scattering angle using
+      // maxPtScattering instead of pt.
+      float pT = m_config.pTPerHelixRadius * std::sqrt(S2 / B2) / 2.;
+      if (pT > m_config.maxPtScattering) {
+        float pTscatter = m_config.highland / m_config.maxPtScattering;
+        pT2scatter = pTscatter * pTscatter;
+      }
+      // convert p(T) to p scaling by sin^2(theta) AND scale by 1/sin^4(theta)
+      // from rad to deltaCotTheta
+      // if deltaTheta larger than allowed scattering for calculated pT, skip
+      if (deltaCotTheta2 - error2 > 0) {
+        if (dCotThetaMinusError2 > pT2scatter * iSinTheta2 *
+                                       m_config.sigmaScattering *
+                                       m_config.sigmaScattering) {
+          continue;
+        }
+      }
+      // A and B allow calculation of impact params in U/V plane with linear
+      // function
+      // (in contrast to having to solve a quadratic function in x/y plane)
+      float Im = std::abs((A - B * rM) * rM);
+
+      if (Im <= m_config.impactMax) {
+        top_valid.push_back(top[t]);
+        // inverse diameter is signed depending if the curvature is
+        // positive/negative in phi
+        curvatures.push_back(B / std::sqrt(S2));
+        impactParameters.push_back(Im);
+      }
+    }
+    if (!top_valid.empty()) {
+      m_config.seedFilter->filterSeeds_2SpFixed(
+          *bottom[b], middle, top_valid, curvatures, impactParameters, Zob, it);
+    }
+  }
+}
+
+template <typename external_spacepoint_t>
+template <std::size_t NDims, typename output_it_t>
+void SeedFinderOrthogonal<external_spacepoint_t>::processM(
```

Comment:
please add a line of comment for each cut applied... or at least for the less obvious ones :)

Reply:
Added!